apps/angular/src/app/features/auth/login/login.component.ts

```
@@ -0,0 +1,94 @@
+@Component({
+	selector: 'camp-login',
+	templateUrl: './login.component.html',
+	styleUrls: ['./login.component.css'],
+})
+export class LoginComponent {
+	/** Show password or not. */
+	protected hidePassword = true;
+
+	/** Show password retype or not. */
+	protected hidePasswordRetype = true;
+
+	private readonly authService = inject(AuthService);
+
+	private readonly router = inject(Router);
+
+	private readonly fb = inject(FormBuilder);
+
+	/** Form group for login. */
+	protected readonly loginForm = this.fb.group(
+		{
+			email: ['', Validators.required],
+			password: ['', Validators.required],
+		},
+		{ updateOn: 'submit' },
+	);
+
+	/**
+	 * Error handler for login request.
+	 * @param error Error.
+	 */
+	private handleLoginError(error: unknown): Observable<never> {
+		if (error instanceof HttpErrorResponse) {
+			error.error.errors.forEach((errorObject: ErrorType) => {
+				const control = this.loginForm.get(errorObject.attr);
+				if (control) {
+					control.setErrors({
+						...(control.errors ?? {}),
+						[errorObject.code]: errorObject.detail,
+					});
+				} else {
+					this.loginForm.controls.email.setErrors({
+						[errorObject.code]: errorObject.detail,
+					});
+					this.loginForm.controls.password.setErrors({
+						[errorObject.code]: errorObject.detail,
+					});
+				}
+			});
+		}
+		return throwError(() => error);
+	}
+
+	/**
+	 * Return errors array of form field.
+	 * @param formField Form field.
+	 */
+	protected getFieldErrors(formField: FormControl<string | null>): string[] {
+		let errorsArray: string[] = [];
+		if (formField.errors) {
+			errorsArray = Object.keys(formField.errors).map(error => formField.getError(error));
+		}
+		return errorsArray;
+	}
+
+	/** Login form submit. */
+	protected onSubmit(): void {
+		if (!this.loginForm.invalid) {
+			const body = this.loginForm.getRawValue();
+			this.authService.login(body)
+				.pipe(
+					first(),
+					catchError((error: unknown) => this.handleLoginError(error)),
```

Silencing an error is not the best option. What if you may have several error handlers? For example, one at the domain layer and one at the UI layer?